static/app/views/settings/featureFlags/changeTracking/newProviderForm.tsx

```diff
@@ -19,78 +19,130 @@ import Form from 'sentry/components/forms/form';
 import TextCopyInput from 'sentry/components/textCopyInput';
 import {t, tct} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
-import {handleXhrErrorResponse} from 'sentry/utils/handleXhrErrorResponse';
-import {useMutation, useQueryClient} from 'sentry/utils/queryClient';
-import type RequestError from 'sentry/utils/requestError/requestError';
+import {useQueryClient} from 'sentry/utils/queryClient';
 import normalizeUrl from 'sentry/utils/url/normalizeUrl';
 import useApi from 'sentry/utils/useApi';
 import {useNavigate} from 'sentry/utils/useNavigate';
 import useOrganization from 'sentry/utils/useOrganization';
-import {makeFetchSecretQueryKey} from 'sentry/views/settings/featureFlags/changeTracking';
+import {
+  makeFetchSecretQueryKey,
+  type Secret,
+} from 'sentry/views/settings/featureFlags/changeTracking';
 
-type CreateSecretQueryVariables = {
-  provider: string;
-  secret: string;
+type UseProviderFormSubmissionProps = {
+  onCreatedSecret: (secret: string) => void;
+  onError: (error: string | null) => void;
+  onSetProvider: (provider: string) => void;
 };
 
-type CreateSecretResponse = string;
+function useProviderFormSubmission({
+  onCreatedSecret,
+  onSetProvider,
+  onError,
+}: UseProviderFormSubmissionProps) {
+  const organization = useOrganization();
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  return useCallback(
+    (
+      data: Record<string, any>,
+      onSubmitSuccess: (response: any) => void,
+      onSubmitError: (error: any) => void
+    ) => {
+      addLoadingMessage();
+
+      api
+        .requestPromise(`/organizations/${organization.slug}/flags/signing-secrets/`, {
+          method: 'POST',
+          data: {
```

Comment:
looks like we were using `useMutation()` before, but now it's a direct api call. It would be ideal to continue with the `useMutation` hook